yt/data_objects/construction_data_containers.py

```diff
@@ -2774,36 +2782,29 @@ def get_data(self, fields=None):
         if fields is None:
             return
 
-        # not sure on the best way to do this
-        if isinstance(fields, list) and len(fields) > 1:
-            for field in fields:
-                self.get_data(field)
-            return
-        elif isinstance(fields, list):
-            fields = fields[0]
+        sph_ptypes = getattr(self.ds, "_sph_ptypes", ["PartType0"])
```

Comment:
Something like this:
```suggestion
        if hasattr(self.ds, "_sph_ptypes"):
            sph_ptypes = self.ds._sph_ptypes
        else:
            sph_ptypes = tuple(value for value in self.ds.particle_types_raw if value in ["PartType0","Gas","gas","io"])
            
        if len(sph_ptypes) ==  0: raise RuntimeError 
```

Reply:
@AshKelly what do you think about the change I suggested?

Reply:
Thanks @jzuhone, this is great! I was a little hesitant to hard code the `["PartType0","Gas","gas","io"]` but I guess these options shouldn't really change.

Reply:
I agree tho--we need to find something better long-term. Should be able to, but for now this is fine.